File: Javascript/이코테_문제풀이/p.110_상하좌우.js
const fs = require("fs");
const input = fs.readFileSync("./Javascript/ex.txt").toString().split("\n");

function solution(input) {
  const n = input[0].trim();
  const plans = input[1].split(" ");
  const directions = {
    L: [0, -1],
    R: [0, 1],
    U: [-1, 0],
    D: [1, 0],
  };
  let [x, y] = [1, 1];
  plans.forEach((plan) => {
    let temp_x = x;
    let temp_y = y;
    temp_x += directions[plan][0];
    temp_y += directions[plan][1];
    if (temp_x > 0 && temp_y > 0 && temp_x <= n && temp_y <= n) {
      x = temp_x;
      y = temp_y;
    }
  });
  return [x, y];
}

console.log(solution(input).join(" "));

File: Python/이코테_문제풀이/p.110_상하좌우.py
# move를 순회하며 L,R,U,D 각각의 경우를 처리해야 한다.
# 공간 밖으로 나갈 경우 무시해야 한다.
# 5
# R R R U D D
n = int(input());
plan = input().split();

x, y = 1, 1;
direction = {
    'L':(0,-1),
    'R':(0,1),
    'U':(-1,0),
    'D':(1,0)
};
def move(key):
    global x,y;
    temp_x = x + direction.get(key)[0];
    temp_y = y + direction.get(key)[1];
    if temp_x > 0 and temp_y > 0 and temp_x <= n and temp_y <= n:
        x = temp_x;
        y = temp_y;

for i in plan:
    move(i);

print(x,y);
